### PR TITLE
Closes #40 Initial ADCOEQ vignette

### LINKED ISSUE #40: Vignette: CoEQ ADaM (ADCOEQ?)
### Background Information

Once #39 is completed, we will have test data for CoEQ. Then we can make a vignette explaining how to categorise/group the data. this may inform whether we want to make specific  new functions or whether existing {admiral} functionality (see [questionnaires user guide](https://pharmaverse.github.io/admiral/dev/articles/questionnaires.html)) is enough.

### Definition of Done

- [ ] Agree dataset name (ADCOEQ?)
- [ ] Create vignette
- [ ] Ensure that right at the top of the vignette we have text to the effect of: _UoLeeds is the license holder and any parties wishing to use the CoEQ questionnaire need to purchase the license (at the appropriate link)_ (precise wording to be agreed with UoLeeds).
- [ ] Make any requests for new {admiralmetabolic} functions.

### PR DESCRIPTION
**Pull request by kathrinflunkert**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
Thanks for this! Some preliminary notes before review (feel free to ignore if still developing):

- You'll need to add an entry in `_pkgdown.yml` so the vignette appears in the navigation bar.
- Title is quite long, what about: Creating a Control of Eating Questionnaire ADaM
- I would make the license text even more prominent - perhaps a "License" section right above the introduction.
- In the intro section, please reference the Questionnaire ADaM vignette (see how ADVS template did it)
- Ensure "admiralmetabolic" is inside curly brackets and backticks (`{admiralmetabolic}`) throughout for correct linkage.
- Ensure "pharmaverse", dataset names and variables are also in backticks - see ADVS vignette.
- `tibble` shouldn't be a required package as you use it only for internal examples
- After "Required Packages", add "Programming Workflow" section - same as ADVS vignette. In general try follow ADVS vignette layout for consistency across the package.
- For Read In Data section, mimic ADVS vignette there too - show any code for qs and dm/adsl
- For unscheduled visits, is this approach likely to vary across studies? Should we specfiy this?
- Not sure baseline and change from baseline sections are needed. Can we just reference admiral documentation?

And the most complex comment:

Can you leverage call_derivation() to make the code in the "scales and scores" section more efficient. I give an example below for two parameters and then you can generalise:
```
adcoeq3 <- adcoeq2 %>%
  call_derivation(
    derivation = derive_summary_records,
    variable_params = list(
      params(
        filter_add = (PARAMCD %in% c('COEQ09','COEQ10','COEQ11','COEQ12','COEQ19')),
        set_values_to = exprs(
          AVAL = mean(AVAL, na.rm = TRUE),
          PARAMCD = "COEQCRCO",
          PARAM = "COEQ - Craving Control"
      ),
      params(
        filter_add = (PARAMCD %in% c('COEQ03','COEQ13','COEQ14','COEQ15')),
         set_values_to = exprs(
           AVAL = mean(AVAL, na.rm = TRUE),
           PARAMCD = "COEQCRSW",
           PARAM = "COEQ - Craving for Sweet"
       )
    ),
    dataset_add = . ,
    by_vars = exprs(STUDYID, USUBJID, AVISIT, AVISITN, ADT, ADY, PARCAT1, TRTSDT, TRTEDT, TRT01P, TRT01A)
```

**Comment by Sonali-netizen** [comment]
Hi Kathrin,
Thanks for putting this up so quickly.
Few comments:

- Title could be "Creating Control of Eating Questionnaire dataset"
- Are we considering imputing baseline values in case the values are missing at baseline visit. Meaning if the participant does not respond to the questionnaire at the baseline visit but responds to the questionnaire at subsequent visits, then do we impute the values? Or leave it as it?
- the derive_summary_records part can be written like a function/macro and called for the different PARAMs
- Is there a requirement of a derived "Total Score" row? I have not worked on this particular questionnaire but in some other questionnaires, there are some derived rows for either a "Total Score" or an "Index Score".
- The BASE, CHG, PCHG variables wouldn't be required unless there are some derived "Total Score" or "Index Score".
- Same for the AVALCATy variables as well.
- Hope towards the end of the code, we just ensure that all the required variables are kept, and other variables are dropped.

**Comment by kathrinflunkert** [comment]
Thanks for your comments, both! 

@manciniedoardo 
I updated the code for most of your comments apart from these:
- "For unscheduled visits, is this approach likely to vary across studies? Should we specfiy this?"
  - I guess it is up to the user how they want to use unscheduled visits - just like in all other ADaMs in general.
- "Not sure baseline and change from baseline sections are needed. Can we just reference admiral documentation?"
  - I agree that we could also refer to the admiral documentation, but then what do we put in this vignette apart from the 4 subscale derivations? I left the code in for now.
- "Can you leverage `call_derivation()` to make the code in the "scales and scores" section more efficient."
  - I tested this approach and the downside is that it does not include the original items anymore, but ADCOEQ should be a combination of original plus derived parameters. That's why I left the code as it was for now.


@Sonali-netizen 
I made a few updates.
To some of your points in particular:
- "Are we considering imputing baseline values in case the values are missing at baseline visit?" 
  - I think imputation of missing data is a general topic and not specific to this questionnaire. 
- "Is there a requirement of a derived "Total Score" row?"
  - The CoEQ does not have an overall Total Score, but only the four derived subscales as I derived them in this code.
- "Hope towards the end of the code, we just ensure that all the required variables are kept, and other variables are dropped."
  - The last dataset in the current code only contains the ADSL variables kept at the beginning, the QS variables and the newly derived ADaM variables.

File: vignettes/adcoeq.Rmd [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+`AVAL`, since that's what will be used for the score calculation.
+For the CoEQ item 6 has to be handled in such way:
+
+```{r eval=TRUE}
+adcoeq2 <- adcoeq1 %>%
+  # Add analysis parameter variables
+  mutate(
+    AVAL = ifelse(PARAMCD=='COEQ06', 100-QSSTRESN, QSSTRESN),
+    AVALC = ifelse(PARAMCD=='COEQ20', QSORRES, NA)
+  )
```
Comment (AndersAskeland) [inline vignettes/adcoeq.Rmd:126]: I would stick with dplyr, add spaces, and use '""' quotes to be consistent.

```suggestion
  mutate(
    AVAL = if_else(PARAMCD == "COEQ06", 100-QSSTRESN, QSSTRESN),
    AVALC = if_else(PARAMCD == "COEQ20", QSORRES, NA_character_)
  ) 
```

File: vignettes/adcoeq.Rmd [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    AVISIT = if_else(ADT <= TRTSDT, "BASELINE", VISIT),
+    AVISITN = if_else(ADT <= TRTSDT, 0, VISITNUM)
+  ) 
+```
+
+If the numeric values of the answers in `QSSTRESN` values are not ready to be used for deriving
+scores and require transformation, it is recommended that `QSSTRESN` is kept in
+the ADaM dataset for traceability, and the transformed value is stored in
+`AVAL`, since that's what will be used for the score calculation.
+For the CoEQ item 6 has to be handled in such way:
```
Comment (AndersAskeland) [inline vignettes/adcoeq.Rmd:118]: Should we specify why item 6 needs to be handled like this?

File: vignettes/adcoeq.Rmd [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+the items. 
+For the Control of Eating questionnaire, four subscales are derived:
+
+* Craving Control: Calculate average of items 9, 10, 11, 12 and 19.
+
+* Craving for Sweet: Calculate average of items 3, 13, 14 and 15.
+
+* Craving for Savoury: Calculate average of items 4, 16, 17 and 18.
+
+* Positive Mood: Calculate average of items 5, 7, 8 and 6 (reversed).
```
Comment (AndersAskeland) [inline vignettes/adcoeq.Rmd:153]: Would be great to have an explanation further up, why we reverse 6.

**Comment by manciniedoardo** [comment]
> Thanks for your comments, both!
> 
> @manciniedoardo I updated the code for most of your comments apart from these:

> * "Can you leverage `call_derivation()` to make the code in the "scales and scores" section more efficient."
>   
>   * I tested this approach and the downside is that it does not include the original items anymore, but ADCOEQ should be a combination of original plus derived parameters. That's why I left the code as it was for now.

What do you mean by this @kathrinflunkert? I think it should be possible to implement as in principle `call_derivation()` is a shorthand for your code so we shouldn't observe differences. Maybe you can share the problematic code? Thanks.

**Comment by yurovska** [comment]
> * "Can you leverage `call_derivation()` to make the code in the "scales and scores" section more efficient."
>   
>   * I tested this approach and the downside is that it does not include the original items anymore, but ADCOEQ should be a combination of original plus derived parameters. That's why I left the code as it was for now.

You need to use a dot in both `dataset` and `dataset_add` to make it work properly. Here is the working example:

```R
adcoeq3 <- adcoeq2 %>%
  call_derivation(
    dataset = .,     # <--- This cannot be omitted if you also use the dot somewhere else
    derivation = derive_summary_records,
    variable_params = list(
      params(
        filter_add = PARAMCD %in% c("COEQ09", "COEQ10", "COEQ11", "COEQ12", "COEQ19"),
        set_values_to = exprs(
          AVAL = mean(AVAL, na.rm = TRUE),
          PARAMCD = "COEQCRCO",
          PARAM = "COEQ - Craving Control"
        )
      ),
      params(
        filter_add = PARAMCD %in% c("COEQ03", "COEQ13", "COEQ14", "COEQ15"),
        set_values_to = exprs(
          AVAL = mean(AVAL, na.rm = TRUE),
          PARAMCD = "COEQCRSW",
          PARAM = "COEQ - Craving for Sweet"
        )
      )
    ),
    dataset_add = .,
    by_vars = exprs(STUDYID, USUBJID, AVISIT, AVISITN, ADT, ADY, PARCAT1, TRTSDT, TRTEDT, TRT01P, TRT01A)
  )
```

Or alternatively you can omit `dataset` argument but use `adcoeq2` explicitly in `dataset_add`.

File: vignettes/adcoeq.Rmd [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+dm <- convert_blanks_to_na(dm_metabolic)
+qs <- convert_blanks_to_na(qs_metabolic)
+admiral_adsl <- convert_blanks_to_na(admiral_adsl)
+```
+
+Within this vignette, `DM` is used as the basis for `ADSL`:
+
+```{r echo=FALSE}
+dm <- admiralmetabolic::dm_metabolic
+
```
Comment (Siddhesh2097) [inline vignettes/adcoeq.Rmd:80]: No need to create dm again here

File: vignettes/adcoeq.Rmd [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    PARCAT1 = QSCAT
+  ) %>%
+  # Add timing variables
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  mutate(
+    AVISIT = if_else(ADT <= TRTSDT, "BASELINE", VISIT),
+    AVISITN = if_else(ADT <= TRTSDT, 0, VISITNUM)
+  ) 
+```
```
Comment (Siddhesh2097) [inline vignettes/adcoeq.Rmd:112]: We can display a vignette here to show how adcoeq1 looks.

File: vignettes/adcoeq.Rmd [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    filter = !is.na(AVAL) & ADT <= TRTSDT
+  ) %>%
+  # Derive baseline and change from baseline variables
+  derive_var_base(
+    by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+    source_var = AVAL,
+    new_var = BASE
+  ) %>%
+  # Calculate CHG for post-baseline records
+  # The decision on how to populate pre-baseline and baseline values of CHG is left to producer choice
```
Comment (Siddhesh2097) [inline vignettes/adcoeq.Rmd:235]:
```suggestion
  # The decision on how to populate pre-baseline and baseline values of CHG is left to producer's choice
```

File: vignettes/adcoeq.Rmd [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+  restrict_derivation(
+    derivation = derive_var_chg,
+    filter = AVISITN > 0
+  ) %>%
+  # Calculate PCHG for post-baseline records
+  # The decision on how to populate pre-baseline and baseline values of PCHG is left to producer choice
```
Comment (Siddhesh2097) [inline vignettes/adcoeq.Rmd:241]:
```suggestion
  # The decision on how to populate pre-baseline and baseline values of PCHG is left to producer's choice
```

File: vignettes/adcoeq.Rmd (hunk 4 above)
Comment (kathrinflunkert) [reply vignettes/adcoeq.Rmd]: True. Removed.

File: vignettes/adcoeq.Rmd [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+           PARAM = "COEQ - Craving for Sweet"
+       )
+      ),
+      params(
+        filter_add = PARAMCD %in% c('COEQ04','COEQ16','COEQ17','COEQ18'),
+         set_values_to = exprs(
+           AVAL = mean(AVAL, na.rm = TRUE),
+           PARAMCD = "COEQCRSA",
+           PARAM = "COEQ - Craving for Savoury"
+       )
```
Comment (yurovska) [inline vignettes/adcoeq.Rmd:180]:
```suggestion
        set_values_to = exprs(
          AVAL = mean(AVAL, na.rm = TRUE),
          PARAMCD = "COEQCRSA",
          PARAM = "COEQ - Craving for Savoury"
        )
```

File: vignettes/adcoeq.Rmd [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+      ),      
+      params(
+        filter_add = PARAMCD %in% c('COEQ05','COEQ07','COEQ08','COEQ06'),
+         set_values_to = exprs(
+           AVAL = mean(AVAL, na.rm = TRUE),
+           PARAMCD = "COEQPOMO",
+           PARAM = "COEQ - Positive Mood"
+       )
```
Comment (yurovska) [inline vignettes/adcoeq.Rmd:188]:
```suggestion
        set_values_to = exprs(
          AVAL = mean(AVAL, na.rm = TRUE),
          PARAMCD = "COEQPOMO",
          PARAM = "COEQ - Positive Mood"
        )
```

**Review by yurovska — CHANGES_REQUESTED** [review]

File: vignettes/adcoeq.Rmd (hunk 2 above)
Comment (kathrinflunkert) [reply vignettes/adcoeq.Rmd]: @AndersAskeland  I added some text about why item 6 needs to be reversed. Please feel free to change the wording.

**Comment by manciniedoardo** [comment]
@kathrinflunkert can we remove draft status now?

File: vignettes/adcoeq.Rmd [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+    AVAL = ifelse(PARAMCD=='COEQ06', 100-QSSTRESN, QSSTRESN),
+    AVALC = ifelse(PARAMCD=='COEQ20', QSORRES, NA)
+  ) 
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adcoeq2, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, QSSTRESN, AVALC, AVAL, ADY, AVISIT)
+)
```
[the remark this answers is not in the document]
Comment (kathrinflunkert) [reply vignettes/adcoeq.Rmd:130]: Is there a way to increase the number of rows shown? If the number of rows is fixed to 10, we would need to filter for items 1 to 9 plus item 20 or something like that.

[the remark this answers is not in the document]
Comment (manciniedoardo) [reply vignettes/adcoeq.Rmd:130]: you could add some filtering to keep only rows you want maybe?

File: vignettes/adcoeq.Rmd [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+          PARAM = "COEQ - Craving Control"
+        )
+      ),
+      params(
+        filter_add = PARAMCD %in% c('COEQ03','COEQ13','COEQ14','COEQ15'),
+         set_values_to = exprs(
+           AVAL = mean(AVAL, na.rm = TRUE),
+           PARAMCD = "COEQCRSW",
+           PARAM = "COEQ - Craving for Sweet"
+       )
```
Comment (yurovska) [inline vignettes/adcoeq.Rmd:172]:
```suggestion
        set_values_to = exprs(
          AVAL = mean(AVAL, na.rm = TRUE),
          PARAMCD = "COEQCRSW",
          PARAM = "COEQ - Craving for Sweet"
        )
```

File: vignettes/adcoeq.Rmd (hunk 10 above)
[the remark this answers is not in the document]
Comment (kathrinflunkert) [reply vignettes/adcoeq.Rmd:138]: Done

File: vignettes/adcoeq.Rmd [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, AVAL, ADY, AVISIT),
+  filter = PARAMCD %in% c("COEQCRCO", "COEQCRSW", "COEQCRSA", "COEQPOMO")
+)
+```
+
+## Remaining ADCOEQ Set-up {#adcoeq_end}
+
+The `{admiral}` [Creating Questionnaire ADaMs vignette](https://pharmaverse.github.io/admiral/articles/questionnaires.html) 
+describes further steps like how to calculate the change from baseline variables or how to add parameters for questionnaire completion.
+
```
Comment (Siddhesh2097) [inline vignettes/adcoeq.Rmd:215]: Should we not add the adcoeq template link here like advs vignette ?

File: vignettes/adcoeq.Rmd [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adcoeq2, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, QSSTRESN, ADY, AVISIT, AVALC, AVAL),
+  filter = PARAMCD %in% c("COEQ01", "COEQ02", "COEQ03", "COEQ04", "COEQ05", "COEQ06", "COEQ07", "COEQ08", "COEQ09", "COEQ20")
+)
+```
+
+We handle unscheduled visits as normal visits. For deriving visits based on
+time-windows, see `{admiral}` [Visit and Period Variables](https://pharmaverse.github.io/admiral/articles/visits_periods.html).
```
Comment (AndersAskeland) [inline vignettes/adcoeq.Rmd:142]:
```suggestion
For this example, we handle unscheduled visits as normal visits, and hence, don't need to perform any derivations. For deriving visits based on
time-windows, see `{admiral}` [Visit and Period Variables](https://pharmaverse.github.io/admiral/articles/visits_periods.html).
```

I am somewhat unsure why we write how we handle unscheduled visits, when we don't actually do anything. I have tried to formulate something, but unsure if it makes sense?

File: vignettes/adcoeq.Rmd [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+
+
+## Derive the four Subscales {#subscales}
+
+Scales and Scores are often derived as the sum or the average across a subset of
+the items. 
+For the Control of Eating Questionnaire, four subscales are derived:
```
Comment (AndersAskeland) [inline vignettes/adcoeq.Rmd:149]:
```suggestion
For the Control of Eating Questionnaire, four subscales are derived. These subscales are derived as the sum or the average across a subset of the various items/questions. 

The subscales are defined as follows:
```

File: vignettes/adcoeq.Rmd [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+
+* Craving Control: Calculate average of items 9, 10, 11, 12 and 19.
+
+* Craving for Sweet: Calculate average of items 3, 13, 14 and 15.
+
+* Craving for Savoury: Calculate average of items 4, 16, 17 and 18.
+
+* Positive Mood: Calculate average of items 5, 7, 8 and 6 (reversed).
```
Comment (AndersAskeland) [inline vignettes/adcoeq.Rmd:157]:
```suggestion
* Positive Mood: Calculate average of items 5, 7, 8 and 6 (inversed).
```

File: vignettes/adcoeq.Rmd [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+  arrange(adcoeq3, USUBJID, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, AVAL, ADY, AVISIT),
+  filter = PARAMCD %in% c("COEQCRCO", "COEQCRSW", "COEQCRSA", "COEQPOMO")
+)
+```
+
+## Remaining ADCOEQ Set-up {#adcoeq_end}
+
+The `{admiral}` [Creating Questionnaire ADaMs vignette](https://pharmaverse.github.io/admiral/articles/questionnaires.html) 
+describes further steps like how to calculate the change from baseline variables or how to add parameters for questionnaire completion.
```
Comment (AndersAskeland) [inline vignettes/adcoeq.Rmd:214]:
```suggestion
The `{admiral}` [Creating Questionnaire ADaMs vignette](https://pharmaverse.github.io/admiral/articles/questionnaires.html) 
describes further steps, including, how to calculate the change from baseline variables, and how to add parameters for questionnaire completion.
```

**Review by AndersAskeland — CHANGES_REQUESTED** [review]
Great work @kathrinflunkert !

It is very close to being mergeable (don't know if that's a word). I have a couple of suggestions for slightly different text formulations, but besides that, its great! 👍

File: vignettes/adcoeq.Rmd (hunk 2 above)
Comment (AndersAskeland) [reply vignettes/adcoeq.Rmd]: Thanks! I think it make sense to have it. I've made a slightly rephrased version in my latest review :)

File: vignettes/adcoeq.Rmd (hunk 13 above)
Comment (manciniedoardo) [reply vignettes/adcoeq.Rmd:142]: I would be in favour of cutting entirely

File: vignettes/adcoeq.Rmd (hunk 15 above)
Comment (manciniedoardo) [reply vignettes/adcoeq.Rmd:157]: Wouldn't inverse indicate "1/"? In which case reverse would be clearer?

File: vignettes/adcoeq.Rmd (hunk 13 above)
Comment (kathrinflunkert) [reply vignettes/adcoeq.Rmd:143]: I would suggest we keep the general reference ("For deriving visits based on time-windows, see ..."), but we remove the first sentence ("We handle unscheduled visits as normal visits."). What do you think?

File: vignettes/adcoeq.Rmd (hunk 12 above)
Comment (manciniedoardo) [reply vignettes/adcoeq.Rmd:216]: Yes please add, as event if the link ends up being broken, we know it will be live in 2 weeks time.

File: vignettes/adcoeq.Rmd (hunk 13 above)
Comment (manciniedoardo) [reply vignettes/adcoeq.Rmd:143]: Agree

File: vignettes/adcoeq.Rmd (hunk 12 above)
Comment (kathrinflunkert) [reply vignettes/adcoeq.Rmd:221]: Done

File: vignettes/adcoeq.Rmd (hunk 13 above)
Comment (kathrinflunkert) [reply vignettes/adcoeq.Rmd]: Done

**Review by yurovska — APPROVED** [review]
Looks good to me! Thanks @kathrinflunkert !

**Review by AndersAskeland — APPROVED** [review]
Looks good to me to! 

As for the warning in the "Man Pages / Roxygen (pull_request) " workflow, I think we can ignore that, as it is triggered because your branch still have an internal assert function (is now removed).

**Comment by Siddhesh2097** [comment]
Looks good to me too. Thanks @kathrinflunkert 👍🏻

**Review by Siddhesh2097 — APPROVED** [review]

**Review by manciniedoardo — APPROVED** [review]
